ironfish-zkp/src/circuits/spend.rs

```diff
@@ -50,6 +54,127 @@ pub struct Spend {
     pub sender_address: Option<SubgroupPoint>,
 }
 
+impl Spend {
+    pub fn write<W: Write>(&self, mut writer: W) -> std::io::Result<()> {
+        if let Some(value_commitment) = &self.value_commitment {
+            writer.write_u8(1)?;
+            writer.write_all(&value_commitment.to_bytes())?;
+        } else {
+            writer.write_u8(0)?;
+        }
+        if let Some(proof_generation_key) = &self.proof_generation_key {
+            writer.write_u8(1)?;
+            writer.write_all(proof_generation_key.to_bytes().as_ref())?;
+        } else {
+            writer.write_u8(0)?;
+        }
+        if let Some(payment_address) = &self.payment_address {
+            writer.write_u8(1)?;
+            writer.write_all(payment_address.to_bytes().as_ref())?;
+        } else {
+            writer.write_u8(0)?;
+        }
+        if let Some(commitment_randomness) = &self.commitment_randomness {
+            writer.write_u8(1)?;
+            writer.write_all(commitment_randomness.to_bytes().as_ref())?;
+        } else {
+            writer.write_u8(0)?;
+        }
+        if let Some(ar) = &self.ar {
+            writer.write_u8(1)?;
+            writer.write_all(ar.to_bytes().as_ref())?;
+        } else {
+            writer.write_u8(0)?;
+        }
+        writer.write_all((self.auth_path.len() as u32).to_le_bytes().as_ref())?;
```

Comment:
Is it possible that `auth_path` does not fit in a `u32`?

Reply:
Seems you are right, this would be a bug. Looking at some of our typescript:
https://github.com/iron-fish/ironfish/blob/ca60831b8556978164e7f66a3d5359a59fbc3609/ironfish/src/primitives/rawTransaction.ts#L284